src/utils/params/pagingParams.ts

```diff
@@ -0,0 +1,47 @@
+import { RequestParams } from './requestParams';
+import { ApiResponseOnSuccess } from '../response/apiResponse';
+
+export class PagingParams<T extends Object> extends RequestParams<T> {
+  private readonly valid: boolean;
+  private readonly startTime: string;
+
+  constructor(params: RequestParams<T> | PagingParams<T>, apiResponse: ApiResponseOnSuccess<T[]>) {
+    super();
+    this.criteria = params.getCriteria();
+
+    let startTime = '';
+    if (params instanceof RequestParams) {
+      startTime = apiResponse.RequestStartTime;
+    }
+
+    if (params instanceof PagingParams) {
+      startTime = params.startTime;
+    }
+
+    this.startTime = startTime;
```

Comment:
We could remove this variable and directly assign to `this.startTime` I think

Reply:
NO, not possible because the compiler will think that it is optional so startTime needs to have "?" then, which I don't like.